operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,315 @@
+}
+
+type LabelKeys struct {
+	// key is the label key.
+	// A label key must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length.
+	// To support older openshift versions, matches is used instead of CEL validation.
+	// The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// The optional prefix and the name are separate by a forward slash (/).
+	// +key uses matches and not cel functions to allow for use on 4.17.
+	// +kubebuilder:validation:MaxLength=317
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self.matches(r'^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?([a-z0-9]([-a-z0-9]*[a-z0-9])?)$')"
+	// +required
+	Key string `json:"key"`
+}
+
+// +kubebuilder:validation:Enum=ByWorkload;None;""
+type GangSchedulingPolicy string
+
+const (
+	GangSchedulingPolicyByWorkload GangSchedulingPolicy = "ByWorkload"
+	GangSchedulingPolicyNone       GangSchedulingPolicy = "None"
+)
+
+// +kubebuilder:validation:Enum="";Parallel;Sequential
+type GangSchedulingWorkloadAdmission string
+
+const (
+	GangSchedulingWorkloadAdmissionSequential GangSchedulingWorkloadAdmission = "Sequential"
+	GangSchedulingWorkloadAdmissionParallel   GangSchedulingWorkloadAdmission = "Parallel"
+)
+
+// Kueue provides the ability to admit workloads all in one (gang admission)
+// and evicts workloads if they are not ready within a specific time.
+// +kubebuilder:validation:XValidation:rule="has(self.policy) && self.policy == 'ByWorkload' ?  has(self.byWorkload) : !has(self.byWorkload)",message="byWorkload is required when policy is byWorkload, and forbidden otherwise"
+// +union
+type GangScheduling struct {
+	// policy allows you to enable and configure gang scheduling.
+	// The allowed values are ByWorkload, None and "".
+	// When set to ByWorkload, this means each workload is processed and considered
+	// for admission as a single unit.
+	// Where workloads do not become ready over time, the entire workload may then be evicted and retried at a later time.
+	// policy is a required field.
+	// +required
+	// +unionDiscriminator
+	Policy GangSchedulingPolicy `json:"policy"`
+	// byWorkload configures how Kueue will process workloads for admission.
+	// byWorkload is required when policy is ByWorkload, and forbidden otherwise.
+	// +optional
+	ByWorkload ByWorkload `json:"byWorkload"`
+}
+
+// ByWorkload controls how admission is done
+type ByWorkload struct {
+	// admission controls how Kueue will process workloads.
+	// admission is a required field with policy is set to ByWorkload.
+	// admission is only required if policy is specified to ByWorkload.
```

Comment:
The user has already set the `byPolicy` object by this point, so these lines aren't adding much